pkg/reconciler/namespace/namespace.go

```diff
@@ -174,6 +178,27 @@ func (r *Reconciler) reconcileServiceAccountAndRoleBindings(ctx context.Context,
 		return fmt.Errorf("track role binding '%s': %v", rb.Name, err)
 	}
 
+	if sa.Name == resources.IngressServiceAccountName || sa.Name == resources.FilterServiceAccountName {
+		// check for existence of brokerPullSecret, and skip copy if it already exists
+		secretFound := false
```

Comment:
I think you can just get rid of this and below when you say secretFound = true, instead of break, just return nil there?

Reply:
this is how I had it before, but @lionelvillard mentioned that perhaps we shouldn't just return nil in the case that someone wants to add more code later on after the copy secret.

Reply:
I'm happy to return nil to keep the code more simple for now, and future folks can add this in easily enough.

Reply:
Yes, it's more golike, so I'd do that. I'm sorry for the runaround, I didn't see those comments, but I'd prefer early return, it's simpler and more golike.

Reply:
Okay, I ended up going with my initial solution here as @vaikas says it's a bit more golike & I agree it's also simpler.